apache-poi/src/main/java/com/baeldung/poi/excel/read/cellvalueandnotformula/CellValueAndNotFormulaHelper.java

```diff
@@ -0,0 +1,81 @@
+package com.baeldung.poi.excel.read.cellvalueandnotformula;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.FormulaEvaluator;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.util.CellAddress;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+public class CellValueAndNotFormulaHelper {
+
+    public Object getCellValueByFetchingLastCachedValue(String fileLocation, String cellLocation) throws IOException {
+        Object cellValue = new Object();
+
+        FileInputStream inputStream = new FileInputStream(new File(fileLocation));
+        Workbook workbook = new XSSFWorkbook(inputStream);
+
+        Sheet sheet = workbook.getSheetAt(0);
+
+        CellAddress cellAddress = new CellAddress(cellLocation);
+        Row row = sheet.getRow(cellAddress.getRow());
+        Cell cell = row.getCell(cellAddress.getColumn());
+
+        if (cell.getCellType() == CellType.FORMULA) {
+            switch (cell.getCachedFormulaResultType()) {
+            case BOOLEAN:
+                cellValue = cell.getBooleanCellValue();
+                break;
+            case NUMERIC:
+                cellValue = cell.getNumericCellValue();
+                break;
+            case STRING:
+                cellValue = cell.getStringCellValue();
+                break;
+            default:
+                cellValue = null;
+            }
+        }
+
+        workbook.close();
+        return cellValue;
+    }
+
+    public Object getCellValueByEvaluatingFormula(String fileLocation, String cellLocation) throws IOException {
+        Object cellValue;
+
+        FileInputStream inputStream = new FileInputStream(new File(fileLocation));
+        Workbook workbook = new XSSFWorkbook(inputStream);
+
+        Sheet sheet = workbook.getSheetAt(0);
+        FormulaEvaluator evaluator = workbook.getCreationHelper()
+            .createFormulaEvaluator();
+
+        CellAddress cellAddress = new CellAddress(cellLocation);
+        Row row = sheet.getRow(cellAddress.getRow());
+        Cell cell = row.getCell(cellAddress.getColumn());
+
+        switch (evaluator.evaluateFormulaCell(cell)) {
```

Comment:
Lets add the same if statement that we have above for clarity
if (cell.getCellType() == CellType.FORMULA) {

Reply:
Added the If condition for better visibility